src/app/app.component.ts

```diff
@@ -1,276 +1,338 @@
+    this.showBadge(UIAction.messageSent);
+    await this.refreshCounter();
+  }
 
-    for (let name of firstNames) {
-      // check if name is on the list already
-      const i = this.mailboxes.filter(m => m.identity.indexOf(name) > -1).length + 1;
-      // add index if it's on the list
-      if (i > 1) {
-        name = `${name} ${i}`;
-      }
-      this.addMailbox(name, null, true);
+  /**
+   * Delete selected messages and files from the relay
+   */
+  async deleteMessages(): Promise<void> {
+    const messagesToDelete = this.activeMailbox.messages.filter(message => message.isSelected);
+    const noncesToDelete = messagesToDelete.map(message => message.nonce);
+    const fileIDsToDelete = messagesToDelete
+      .filter(message => message.kind === ZaxMessageKind.file)
+      .map((message: ZaxFileMessage) => message.uploadID);
+
+    // Delete files before deleting file messages
+    await this.activeMailbox.connectToRelay(this.relayURL);
+    for (const uploadID of fileIDsToDelete) {
+      await this.activeMailbox.deleteFile(this.relayURL, uploadID);
     }
-    this.refreshCounter();
+
+    // Delete file messages
+    await this.activeMailbox.delete(this.relayURL, noncesToDelete);
+
+    // Update messages list and counter without polling server
+    this.activeMailbox.messages = this.activeMailbox.messages.filter(message => !message.isSelected);
+    this.activeMailbox.counter = Object.keys(this.activeMailbox.messages).length;
   }
 
   // -------------------------
-  // Glow operations
+  // File operations
   // -------------------------
 
-  addPublicKey(mailbox, form: NgForm) {
-    const { name, key } = form.controls;
-    if (mailbox.keyRing.addGuest(name.value, key.value)) {
-      this.keyAdded = true;
-      setTimeout(() => {
-        this.keyAdded = false;
-      }, 3000);
+  /**
+   * Send an encrypted file to a specified guest. Only single chunk is allowed for simplicity
+   */
+  async sendFile(guest: string): Promise<void> {
+    if (!this.selectedFile) {
+      return;
     }
-    form.reset();
-  }
 
-  async refreshCounter() {
-    if (!this.mailboxes.length) {
+    const binary = new Uint8Array(await this.selectedFile.arrayBuffer());
+    const metadata = {
+      name: this.selectedFile.name,
+      orig_size: this.selectedFile.size,
+      modified: this.selectedFile.lastModified
+    };
+
+    await this.activeMailbox.connectToRelay(this.relayURL);
+    const { skey, uploadID, max_chunk_size } =
+      await this.activeMailbox.startFileUpload(this.relayURL, guest, metadata);
+
+    if (this.selectedFile.size >= max_chunk_size) {
+      alert(`Error, maximum file size is ${max_chunk_size} bytes`);
       return;
     }
-    this.showRefreshLoader = true;
-    for (const mbx of this.mailboxes) {
-      await this.messageCount(mbx);
-    }
-    this.showRefreshLoader = false;
+
+    await this.activeMailbox.uploadFileChunk(this.relayURL, uploadID, binary, 0, 1, skey);
```

Comment:
so we upload file with size not bigger then 1 chunk?

Reply:
I think it's fine to limit files to 500kb: as I wrote in a comment above, `Only single chunk is allowed for simplicity`. Do you think it makes sense to allow any files? I'm afraid it would become easy to spam Zax with huge files then